### PR TITLE
Vf valuefinder ui

### PR DESCRIPTION
**Pull request by saurabhB-Webonise**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -151,6 +151,7 @@ dependencies {
     implementation "com.google.dagger:dagger:$rootProject.dagger2Version"
     implementation "com.google.dagger:dagger-android:$rootProject.dagger2Version"
     implementation "com.google.dagger:dagger-android-support:$rootProject.dagger2Version"
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (tejasAlsi-Webonise) [inline app/build.gradle:154]: why is this required

File: app/src/main/java/com/sjana/fitness/model/HistoryContentModel.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.sjana.fitness.model
+
+data class HistoryContentModel(
+            var date: String,
+            var list : MutableList<ValueCardsIncluded>,
+            var isChecked:Boolean
+    )
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/model/HistoryContentModel.kt:7]: EOL

File: app/src/main/java/com/sjana/fitness/model/ValueCardsResponse.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+)
+
+data class ValueCardsAttributesX(
+        @SerializedName("definition")
+        val definition: String,
+        @SerializedName("id")
+        val id: Int,
+        @SerializedName("value")
+        val value: String
+)
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/model/ValueCardsResponse.kt:77]: EOL

File: app/src/main/java/com/sjana/fitness/utils/CommonUtils.kt [hunk 4]
```diff
@@ -160,4 +160,8 @@ object CommonUtils {
         val result: Date = cal.getTime()
         return  dateFormat.format(result)
     }
+
+    fun getDateformat(dateString:String): String{
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/utils/CommonUtils.kt:164]: no need for separate function, this is feature specific.

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/TopTenSelectedValueFragment.kt [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            it.findNavController().navigate(action)
+        })
+    }
+
+    interface  IFragmentCallback:Serializable{
+        fun onBeginClick(count:Int,list:MutableList<ValueCardData>)
+    }
+
+
+}
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/TopTenSelectedValueFragment.kt:53]: EOL

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderCompletedFragment.kt [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+        shareVfCTxt.setOnClickListener({
+        })
+
+        historyTxt.setOnClickListener({
+            it.findNavController().navigate(R.id.valueFinderCompletedTovalueFinderHistory)
+        })
+    }
+
+}
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderCompletedFragment.kt:54]: EOL

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt [hunk 7]
```diff
@@ -30,22 +36,51 @@ class ValueFinderFragment : BaseFragment<ValueFinderViewModel>(), ValueFinderNav
...
+    private val args: ValueFinderFragmentArgs by navArgs()
+
+    private var count: Int = 10
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                               savedInstanceState: Bundle?): View? {
         valueFinderViewModel.navigator = this
-        valueFinderViewModel.getValueFinderCards()
+
+        try {
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt:51]: handle this with if statement

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt [hunk 8]
```diff
@@ -54,40 +89,124 @@ class ValueFinderFragment : BaseFragment<ValueFinderViewModel>(), ValueFinderNav
...
         initCardViewPager()
+        handleArgs()
         backArrowVfImg.setOnClickListener({
             findNavController().navigateUp()
         })
     }
 
-    override fun openNextActivity() {
+    fun handleArgs() {
+        try {
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt:188]: no need of try catch

File: app/build.gradle (hunk 1 above)
Comment (saurabhB-Webonise) [reply app/build.gradle:154]: done

File: app/src/main/java/com/sjana/fitness/model/HistoryContentModel.kt (hunk 2 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/model/HistoryContentModel.kt:7]: done

File: app/src/main/java/com/sjana/fitness/model/ValueCardsResponse.kt (hunk 3 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/model/ValueCardsResponse.kt:77]: done

File: app/src/main/java/com/sjana/fitness/utils/CommonUtils.kt (hunk 4 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/utils/CommonUtils.kt:164]: done

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/TopTenSelectedValueFragment.kt (hunk 5 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/TopTenSelectedValueFragment.kt:53]: done

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderCompletedFragment.kt (hunk 6 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderCompletedFragment.kt:54]: done

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderHistoryAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+                holder.itemView.historylayout.visibility = View.VISIBLE
+            }
+            notifyDataSetChanged()
+        })
+    }
+
+     private fun setDrawable(view:TextView,resource:Int){
+         view.setCompoundDrawablesWithIntrinsicBounds(0,0,resource,0)
+     }
+}
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderHistoryAdapter.kt:89]: EOL

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderHistoryFragment.kt [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    override fun setupViews() {
+    }
+
+    override fun openNextActivity() {
+    }
+
+
+}
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderHistoryFragment.kt:86]: EOL

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt [hunk 11]
```diff
@@ -38,6 +41,14 @@ class ValueFinderPagerFragment(valueCardData: ValueCardData,currentPageNo:Int,to
         titleVcTxt.text = mValueCardData.attributes.value
         contentVctxt.text = mValueCardData.attributes.definition
         pageNoVcTxt.text="$mCurrentPageNo / $mTotalPage"
+        completeCardVfBtn.isChecked=mValueCardData.isChecked
+
+
+        Log.e("pager","onViewCreated")
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt:47]: remove this log

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+                    app:layout_constraintTop_toTopOf="parent" />
+
+                <TextView
+                    android:id="@+id/contentFirstTxt"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:gravity="center"
+                    android:letterSpacing="0.05"
+                    android:lineSpacingExtra="3sp"
+                    android:text="You take responsibility for your actions and their consequences, and you respect others who do the same."
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:71]: move this to string.xml

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+    <TextView
+        android:id="@+id/valueDatetxt"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerInParent="true"
+        android:layout_marginTop="25dp"
+        android:drawableRight="@drawable/expand_arrow"
+        android:drawablePadding="10dp"
+        android:fontFamily="@font/heebo_bold"
+        android:text="June 15, 2020"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:17]: remove this

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 14]
```diff
@@ -0,0 +1,231 @@
...
+
+                <TextView
+                    android:id="@+id/titleFirstTxt"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="14dp"
+                    android:fontFamily="@font/heebo_bold"
+                    android:gravity="center_horizontal"
+                    android:letterSpacing="0.05"
+                    android:text="Accountability"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:57]: string.xml

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 15]
```diff
@@ -0,0 +1,231 @@
...
+
+                <TextView
+                    android:id="@+id/titleSecondTxt"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="12dp"
+                    android:fontFamily="@font/heebo_bold"
+                    android:gravity="center"
+                    android:letterSpacing="0.05"
+                    android:text="Bravery"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:86]: remove this

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 16]
```diff
@@ -0,0 +1,231 @@
...
+
+                <TextView
+                    android:id="@+id/contentSecondTxt"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:fontFamily="@font/heebo_regular"
+                    android:gravity="center"
+                    android:letterSpacing="0.05"
+                    android:lineSpacingExtra="3sp"
+                    android:text="Taking action on something even when nervous or scared is something that defines you."
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:101]: string.xml

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+
+                <TextView
+                    android:id="@+id/titleThirdTxt"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="12dp"
+                    android:fontFamily="@font/heebo_bold"
+                    android:gravity="center"
+                    android:letterSpacing="0.05"
+                    android:text="Cleanliness"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:116]: remove this

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 18]
```diff
@@ -0,0 +1,231 @@
...
+
+                <TextView
+                    android:id="@+id/contentThirdTxt"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:fontFamily="@font/heebo_regular"
+                    android:gravity="center"
+                    android:letterSpacing="0.05"
+                    android:lineSpacingExtra="3sp"
+                    android:text="Being organized and keeping your spaces in order are of high value to you and you look for that in others."
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:131]: remove this

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 19]
```diff
@@ -0,0 +1,231 @@
...
+                    android:textColor="@color/darkTxtColor"
+                    android:textSize="12sp"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toBottomOf="@+id/titleThirdTxt" />
+
+                <TextView
+                    android:id="@+id/titleFourthTxt"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:139]: give some meaningful id name

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 20]
```diff
@@ -0,0 +1,231 @@
...
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="12dp"
+                    android:fontFamily="@font/heebo_bold"
+                    android:gravity="center"
+                    android:letterSpacing="0.05"
+                    android:text="Cooperation"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:146]: no hard coded strings

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 21]
```diff
@@ -0,0 +1,231 @@
...
+
+                <TextView
+                    android:id="@+id/contentFourthTxt"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:fontFamily="@font/heebo_regular"
+                    android:gravity="center_horizontal"
+                    android:letterSpacing="0.05"
+                    android:lineSpacingExtra="3sp"
+                    android:text="You thrive in team work environments. You love seeing people bring their own experiences/perspectives to the table."
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:161]: same here

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 22]
```diff
@@ -0,0 +1,231 @@
...
+
+                <TextView
+                    android:id="@+id/titleFifthTxt"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="12dp"
+                    android:fontFamily="@font/heebo_bold"
+                    android:gravity="center"
+                    android:letterSpacing="0.05"
+                    android:text="Cooperation"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:177]: here as well

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 23]
```diff
@@ -0,0 +1,231 @@
...
+
+                <TextView
+                    android:id="@+id/contentFifthTxt"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:fontFamily="@font/heebo_regular"
+                    android:gravity="center_horizontal"
+                    android:letterSpacing="0.05"
+                    android:lineSpacingExtra="3sp"
+                    android:text="You thrive in team work environments. You love seeing people bring their own experiences/perspectives to the table."
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:192]: this one too

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 24]
```diff
@@ -0,0 +1,231 @@
...
+        </androidx.cardview.widget.CardView>
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_horizontal"
+            android:layout_marginTop="10dp"
+            android:drawableLeft="@drawable/ic_share_icon_black"
+            android:fontFamily="@font/heebo_bold"
+            android:gravity="center"
+            android:text="Share"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:219]: this one as well

File: app/src/main/res/layout/recyclerview_cell.xml [hunk 25]
```diff
@@ -0,0 +1,231 @@
...
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/answer_layout" />
+    </LinearLayout>
+
+
+
+
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/recyclerview_cell.xml:231]: EOL

File: app/src/main/res/layout/top_ten_selected_value_fragment.xml [hunk 26]
```diff
@@ -0,0 +1,115 @@
...
+        android:layout_marginLeft="41dp"
+        android:layout_height="50dp"
+        android:layout_marginTop="20dp"
+        android:textSize="22sp"
+        android:textColor="@color/colorWhite"
+        android:gravity="center"
+        android:text="@string/begin"
+        />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/top_ten_selected_value_fragment.xml:115]: EOL

**Review by tejasAlsi-Webonise — CHANGES_REQUESTED** [review]
check comments and remove hard coded strings. Also provide some meaning full names to the id.

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt (hunk 7 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt:51]: done

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt (hunk 8 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt:188]: done

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderHistoryAdapter.kt (hunk 9 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderHistoryAdapter.kt:89]: done

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderHistoryFragment.kt (hunk 10 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderHistoryFragment.kt:86]: done

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt (hunk 11 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt:47]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 12 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:71]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 13 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:17]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 14 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:57]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 15 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:86]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 17 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:116]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 18 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:131]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 20 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:146]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 21 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:161]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 22 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:177]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 23 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:192]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 24 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:219]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 25 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:231]: done

File: app/src/main/res/layout/top_ten_selected_value_fragment.xml (hunk 26 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/top_ten_selected_value_fragment.xml:115]: done

File: app/src/main/res/layout/recyclerview_cell.xml (hunk 19 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/recyclerview_cell.xml:139]: done

**Review by tejasAlsi-Webonise — APPROVED** [review]
